File: backer/godotask/model/process_optimization.go
package model

import (
	"time"
	"gorm.io/gorm"
)

// ProcessOptimization - プロセス最適化記録
type ProcessOptimization struct {
	ID              string    `gorm:"type:varchar(255);primaryKey" json:"id"`
	TaskID        int            `json:"task_id" gorm:"index"`
	ProcessID       string    `json:"process_id" gorm:"index"`
	OptimizationType string   `json:"optimization_type"` // speed, accuracy, energy, cost
	InitialState    JSON      `json:"initial_state" gorm:"type:jsonb"`
	OptimizedState  JSON      `json:"optimized_state" gorm:"type:jsonb"`
	Improvement     float64   `json:"improvement"` // 改善率（%）
	Method          string    `json:"method" gorm:"type:text"`
	Iterations      int       `json:"iterations"`
	ConvergenceTime float64   `json:"convergence_time"` // 収束時間（秒）
	ValidatedBy     string    `json:"validated_by"`
	ValidationDate  time.Time `json:"validation_date"`
	CreatedAt       time.Time `json:"created_at"`
	UpdatedAt       time.Time `json:"updated_at"`
	DeletedAt       gorm.DeletedAt `gorm:"index" json:"deleted_at,omitempty"`
}

File: fronter/src/model/processOptimization.ts
export interface ProcessOptimization {
  id: string;
  process_id: string;
  optimization_type: string; // "speed" | "accuracy" | "energy" | "cost" として列挙型化も可能
  initial_state: any; // JSONB → any
  optimized_state: any; // JSONB → any
  improvement: number; // 改善率（%）
  method: string;
  iterations: number;
  convergence_time: number; // 収束時間（秒）
  validated_by: string;
  validation_date: string; // ISO8601文字列 (time.Time → string)
  created_at: string;
  updated_at: string;
  deleted_at?: string | null; // nullable timestamp
}

export interface AddProcessOptimization {
  process_id: string;
  optimization_type: string; // "speed" | "accuracy" | "energy" | "cost" として列挙型化も可能
  initial_state: any; // JSONB → any
  optimized_state: any; // JSONB → any
  improvement: number; // 改善率（%）
  method: string;
  iterations: number;
  convergence_time: number; // 収束時間（秒）
  validated_by: string;
  validation_date: string; // ISO8601文字列 (time.Time → string)
}
